### PR TITLE
Daniel kuperman/tsk 4177 named triggers for GitHub - cont

### PR DESCRIPTION
**Pull request by blink-daniel**

### CODE REVIEW

File: plugins/github/mask.yaml [hunk 1]
```diff
@@ -34,13 +34,567 @@ triggers:
   - name: OnNewPullRequest
     display_name: On New Pull Request
     description: Triggers a workflow on every new pull request opened in a given repository.
-    action: github.ListPullRequests
+    action: github.SearchIssuesAndPullRequests
     action_params:
       const_values:
         auto_pagination: true
```
Comment (marakomer) [inline plugins/github/mask.yaml:40]: Why use autopagination? 
Could be a bit excessive if done every few minutes

File: plugins/github/mask.yaml [hunk 2]
```diff
@@ -564,6 +1118,13 @@ triggers:
           "draft": false
         }
 actions:
+  search/issues-and-pull-requests:
+    alias: SearchIssuesAndPullRequests
+    parameters:
+      q:
+        alias: "Query"
+      sort:
```
Comment (marakomer) [inline plugins/github/mask.yaml:1126]: Maybe alias to "Sort" and "Order" to be more presentable? Not sure if you want this action to be exposed or only for the trigger

**Review by marakomer — COMMENTED** [review]
Let me know what you think :)

File: plugins/github/mask.yaml (hunk 2 above)
Comment (DanielDinsh) [reply plugins/github/mask.yaml:1126]: If you look in and named trigger that generated - it's Order and Sort

File: plugins/github/mask.yaml (hunk 1 above)
Comment (blink-daniel) [reply plugins/github/mask.yaml:40]: we don't want to miss out events, if we don't we might not run the workflow on a merge/open PR event.

File: components/openapi/trigger.go [hunk 3]
```diff
@@ -24,8 +25,9 @@ type Trigger struct {
 }
 
 type ActionParams struct {
-	ConstValues    map[string]string `yaml:"const_values"`
-	UserParameters map[string]string `yaml:"user_parameters"`
+	ConstValues    map[string]string              `yaml:"const_values"`
+	UserParameters map[string]string              `yaml:"user_parameters"`
+	NewUserParams  []content.NamedActionParameter `yaml:"new_user_parameters"`
```
Comment (DanielDinsh) [inline components/openapi/trigger.go:30]: Maybe change it to something like -
"custom_parameters" or "trigger_custom_parameters" or "helper_parameters" or something like this? 

Just a suggestion, tell me what you think.

Comment (blink-daniel) [reply components/openapi/trigger.go:30]: helper sounds good

**Review by DanielDinsh — APPROVED** [review]

**Review by DanielDinsh — APPROVED** [review]

**Review by marakomer — APPROVED** [review]

**Pull request merged**